test/Microsoft.AspNetCore.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -590,7 +591,7 @@ private class Address2
             Assert.NotNull(boundPerson);
 
             Assert.False(modelState.IsValid);
-            Assert.Equal(1, modelState.Keys.Count);
+            Assert.Equal(1, modelState.Keys.Count());
```

Comment:
Why not assert `modelState.Count` or remove the lambda from the `Assert.Single()` call here too? (Could also have `KeyEnumerable` and `ValueEnumerable` expose `Count => _dictionary.Count` but I bet this is only a testing concern.)